pyserini/search/lucene/tprob/__main__.py

```diff
@@ -0,0 +1,135 @@
+#
+# Pyserini: Reproducible IR research with sparse and dense representations
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import argparse
+import json
+import os
+import subprocess
+from pyserini.search.lucene.tprob import TranslationProbabilitySearcher
+from typing import List
+
+
+def normalize(scores: List[float]):
+    low = min(scores)
+    high = max(scores)
+    width = high - low
+    if width != 0:
+        return [(s-low)/width for s in scores]
+    else:
+        return scores
+
+
+def query_loader(query_path: str):
+    queries = {}
+    with open(query_path) as f:
+        for line in f:
+            query = json.loads(line)
+            qid = query.pop('id')
+            query['analyzed'] = query['analyzed']
+            query['text'] = query['text']
+            query['raw'] = query['raw']
+            query['text_unlemm'] = query['text_unlemm']
+            query['text_bert_tok'] = query['text_bert_tok']
+            queries[qid] = query
+    return queries
+
+
+def sort_dual_list(pred: List[float], docs: List[str]):
+    zipped_lists = zip(pred, docs)
+    sorted_pairs = sorted(zipped_lists)
+
+    tuples = zip(*sorted_pairs)
+    pred, docs = [list(tuple) for tuple in tuples]
+
+    pred.reverse()
+    docs.reverse()
+    return pred, docs
+
+
+def evaluate(qrels_path: str, run_path: str, options: str = ''):
+    curdir = os.getcwd()
+    if curdir.endswith('scripts'):
+        anserini_root = '../../anserini'
+    else:
+        anserini_root = '../anserini'
+    prefix = f"{anserini_root}/tools/eval/trec_eval.9.0.4/trec_eval \
+                -c -M1000 -m all_trec {qrels_path}"
+    cmd1 = f"{prefix} {run_path} {options} | grep 'ndcg_cut_20 '"
+    cmd2 = f"{prefix} {run_path} {options} | grep 'map                   	'"
+    ndcg_string = str(subprocess.check_output(cmd1, shell=True))
+    ndcg_score = ndcg_string.split('\\t')[-1].split('\\n')[0]
+    map_string = str(subprocess.check_output(cmd2, shell=True))
+    map_score = map_string.split('\\t')[-1].split('\\n')[0]
+    return str(map_score), str(ndcg_score)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description='use ibm model 1 feature to rerank the base run file')
+    parser.add_argument('-tag', type=str, default="ibm",
+                        metavar="tag_name", help='tag name for resulting Qrun')
+    parser.add_argument('-qrels', type=str, default="./tools/topics-and-qrels/qrels.msmarco-passage.dev-subset.txt",
```

Comment:
Sorry, should have been more specific. I believe we use double dash everywhere else in Pyserini, so can we switch to double dash to be consistent?

Reply:
No worries, they are all changed to double dash now. Ready to merge.